src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -59,6 +61,38 @@ public static int getFileSizeInKB(String filePath) {
         return Math.toIntExact((file.length() / BYTES_PER_KILOBYTES) + 1L); // Add one so that integer division rounds up
     }
 
+    /**
+     * This method retrieves a specified field mapping from a set of mapping properties.
+     *
+     * @param properties A map representing properties, where each key is a property name and
+     *                   the value is either a sub-map of properties or the property value itself.
+     * @param fieldPaths The field path list that make up the path to the field mapping.
+     * @return           The value of the field if found, or null if the field is not present in the map.
+     */
+    public static Object getFieldMapping(final Map<String, Object> properties, final String[] fieldPaths) {
```

Comment:
Can you search if similar code exists in OpenSearch core that we could re-use instead of implementing on our own?

Also, this function comment isnt quite clear to me. Can you add more details in the comment?

Reply:
Updated function comment

Reply:
Updated the function from feedback